scipy/interpolate/_rgi_cython.pyx

```diff
@@ -0,0 +1,126 @@
+# cython: language_level=3
+"""
+Cythonized routines for the RegularGridInterpolator.
+"""
+
+from libc.math cimport NAN
+import numpy as np
+cimport numpy as np
+cimport cython
+
+include "_poly_common.pxi"
+
+np.import_array()
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+@cython.initializedcheck(False)
+def evaluate_linear_2d(double_or_complex[:, :] values, # cannot declare as ::1
+                       long[:, :]indices,              # unless prior
+                       double[:, :] norm_distances,    # np.ascontiguousarray
+                       tuple grid not None,
```

Comment:
I'm not 100% sure about this. `grid` is a tuple of numpy arrays, right? So we could have them all packed into a array rather than a tuple.

Reply:
Different axes may different sizes, so that `np.asarray(grid)` would be ragged: consider `grid=(np.arange(5), np.arange(6))`. What can be done though, we could pad the arrays and pass the sizes separately. Is this what you meant?

Reply:
I didn't remember that they were different sizes. That makes sense. I'm not too sure if padding the arrays will be that beneficial, but we can experiment with that later